500-CNAB-dependencies.md

```diff
@@ -64,9 +65,14 @@ This section is a placeholder and will be completed in a follow-up pull request.
 This specification introduces a `dependencies` object in the bundle.json
 that defines metadata necessary to specify a dependency.
 
-The entry `dependencies` in the custom extension map, `custom`, is reserved and
+The entry `dependencies` in the custom extension object, `custom`, is reserved and
 MUST only be used for this CNAB Dependencies Specification.
 
+### sequence
+
+The `sequence` array defines the criteria for sequencing bundles in a pre-defined order. The sequence for
```

Comment:
Do we want to entertain stricter verbiage here?  Something like, 'If the sequence array is non-empty, the runtime MUST execute each dependency according to this pre-defined order'?

And/or, if we don't wish to make it a MUST, perhaps we should update to specify SHOULD?

Reply:
Yes, I would agree with `SHOULD` to give flexibility. How about : 
> The `sequence` array defines the criteria for sequencing dependant bundles, if the `sequence` array is non-empty, the runtime SHOULD execute each dependency according to this pre-defined order. 
The sequence for each occurrence is a way for the bundle to reference the order of a dependency.
Also, do we have to indicate that number of the `sequences` has to mutch the number of `requires` to perform the ordering? Otherwise, we handle only the `requires`.

Reply:
I am concerned about the looser SHOULD. Is that intended to allow a runtime to resolve the dependency graph and determine the execution order of the dependencies itself? Or to allow a runtime to opt out of executing dependencies in any reliable order at all?

We can't allow the spec to say it's okay in the second case to ignore the author and run the dependencies like they are today, unordered. An author needs to be able to rely on the dependencies being run in the proper order, otherwise it can unexpectedly fail while executing because the order is incorrect.

For the first, where we want to allow for smarter runtimes but disallow ignoring ordering entirely, how about something like this?

> The sequence array is a pre-defined sequencing of the dependant bundles. If the sequence array is non-empty, the runtime MUST execute each dependency according to this pre-defined order unless the runtime is capable of resolving the dependency graph to determine an execution order itself. The sequence for each occurrence is a way for the bundle to reference the order of a dependency.

Reply:
Yesss ... I love it :)